app/bot.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+import datetime as dt
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, cast, final, override
+
+import discord as dc
+from discord.ext import commands
+from loguru import logger
+
+from app.errors import handle_error, interaction_error_handler
+from app.status import BotStatus
+from app.utils import is_mod, try_dm
+
+if TYPE_CHECKING:
+    from githubkit import GitHub, TokenAuthStrategy
+
+    from app.config import Config, WebhookFeedType
+
+
+@final
+class GhosttyBot(commands.Bot):
+    def __init__(self, config: "Config", gh: "GitHub[TokenAuthStrategy]") -> None:
+        intents = dc.Intents.default()
+        intents.members = True
+        intents.message_content = True
+        super().__init__(
+            command_prefix=commands.when_mentioned_or("!"),
+            intents=intents,
+            allowed_mentions=dc.AllowedMentions(everyone=False, roles=False),
+        )
+
+        self.tree.on_error = interaction_error_handler
+        self.config = config
+        self.gh = gh
+        self.ghostty_status = BotStatus()
+        self.background_tasks = set[asyncio.Task[None]]()
+
+    @override
+    async def on_error(self, event_method: str, /, *args: Any, **kwargs: Any) -> None:
+        handle_error(cast("BaseException", sys.exception()))
+
+    @override
+    async def setup_hook(self) -> None:
+        extensions = (
+            f"app.components.{file.stem}"
+            for file in (Path(__file__).parent / "components").iterdir()
+            if not file.name.startswith("_")
+        )
+        await asyncio.gather(*map(self.load_extension, extensions))
+
+    async def on_ready(self) -> None:
+        # Creating a strong reference
```

Comment:
This doesn't refer to anything, I'm assuming this is supposed to be near the end of `app/components/github_integration/__init__.py`?